spec/betterc.dd

```diff
@@ -0,0 +1,48 @@
+Ddoc
+
+$(SPEC_S Better C,
+
+`-betterC` is a Command-Line flag for `dmd`, which generated light-weight,
+barebone, runtime-free binaries with are suitable for linking from C.
+
+---
+void main()
+{
+    import core.stdc.stdio : printf;
+    printf("Hello betterC");
+}
+---
+
+$(CONSOLE
+> dmd -betterC hello.d
+Hello betterC
+)
+
+$(H2 $(LNAME2 effects, Effects))
+
+    $(OL
+        $(LI Doesn't link DRuntime and standard library by default)
+	    $(LI Doesn't generate ModuleInfo entries)
+    )
+
+$(H2 $(LNAME2 consequences, Consequences))
+
+As no DRuntime is available, many D features won't work. For example:
+
+	$(OL
+	    $(LI No static module constructors or deconstructors)
+	    $(LI No Garbage Collection)
+	    $(LI No dynamic arrays and associative arrays)
+	    $(LI No Classes)
+	    $(LI No `synchronized`, monitors, mutex from $(MREF core, sync))
+	    $(LI No `shared` (`__gshared` can be used instead))
+	    $(LI No $(MREF core, thread))
+	    $(LI No built-in instrinsics (e.g. $(MREF core, math)))
```

Comment:
No, `extern(C)` has nothing to do with that. The problem is that `core.math` functions use `real` (which may be 64 or 80-bit in size) while `printf` 's `%f` format assumes that its argument is `32-bit`. You either need to cast the result to `float`:
```D
void main()
{
    import core.stdc.stdio : printf;
    import core.math : cos;
    printf("Hello world: %.2f\n", cast(float)cos(0));
}
```

```sh
$ dmd -betterC use_core_math.d && ./use_core_math
Hello world: 1.00
```

Or you need to use a different [conversion format specifier](http://en.cppreference.com/w/cpp/io/c/fprintf) like `%Lf`:
```D
void main() {
    import core.stdc.stdio : printf;
    import core.math : cos;
    printf("Hello world: %.2Lf\n", cos(0));
}
```

```sh
$ dmd -betterC use_core_math.d && ./use_core_math
Hello world: 1.00
```

Though, since it's not clear what the size of `real` would be on the target libc, it's better to not rely on that. I think Ilya also avoids using `real`, for performance reasons (SSE is faster than x87) and for precision (avoiding unnecessary rounding when going from FP register to memory and back), etc.